internal/statesync/reactor.go

```diff
@@ -359,39 +363,47 @@ func (r *Reactor) Sync(ctx context.Context) (sm.State, error) {
 	// We need at least two peers (for cross-referencing of light blocks) before we can
 	// begin state sync
 	if err := r.waitForEnoughPeers(ctx, 2); err != nil {
-		return sm.State{}, err
+		return sm.State{}, fmt.Errorf("wait for peers: %w", err)
 	}
 
 	r.mtx.Lock()
```

Comment:
it is not related to your changes, but what do you think if to move mutex usage to a separate method?